### PR TITLE
Added Layer Normalization paper

### PR DESCRIPTION
**Pull request by ppoulin91**

### CODE REVIEW

File: deep-learning/_posts/2017-03-27-layer-normalization.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+---
+layout: review
+title: Layer Normalization
+tags: deep-learning essentials
```
Comment (lemairecarl) [inline deep-learning/_posts/2017-03-27-layer-normalization.md]: Please add the tag `layers`. (The review of Batch Norm should have it too, I thought it had it)

File: deep-learning/_posts/2017-03-27-layer-normalization.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+$$ \mu^l = \frac{1}{H} \sum\limits_{i=1}^H a_i^l  \qquad  \sigma^l = \sqrt{\frac{1}{H} \sum\limits_{i=1}^H (a_i^l - \mu^l)^2} $$
+
+where $$H$$ is the number of hidden units in the layer.
+
+Results of a unidirectional attentive reader model trained on the CNN corpus question-answering task: 
+![](/deep-learning/images/layer-normalization/layer_norm_fig2.png)
+
+
+# Limitations
+**Speed**: While layer normalization will speed up training in terms of epochs/training steps, it will also slow down training in terms of time per epoch/update, because moment computation is much slower on GPU compared to simple matrix multiplications. Overall, it should still usually speed up global training time, and hopefully give better results, but there is no guarantee.
```
Comment (lemairecarl) [inline deep-learning/_posts/2017-03-27-layer-normalization.md]: 1. I suggest replacing "speed up training in terms of epochs/training steps" by "speed up training in terms of **the number of** epochs/training steps"
2. Is there a reason why you use "epochs/training steps" then "epoch/update" ?

File: deep-learning/_posts/2017-03-27-layer-normalization.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+Layer Normalization is a simple method used to reduce training time in deep neural networks. It can also stabilize the hidden state dynamics in recurrent networks. It is similar to Batch Normalization, but is not dependent on mini-batch size, and performs the same computation at training and testing time.
+
+
+# How it works
+Layer normalization works by computing the normalization statistics (mean and variance) from all of the summed inputs to the neurons within a hidden layer (on a single training case).
+
+# Background
+
+A feed-forward neural network is a non-linear mapping from a input pattern $$\mathbf x$$ to an output vector $$y$$.
```
Comment (Dref360) [inline deep-learning/_posts/2017-03-27-layer-normalization.md]: Is it really necessary? Why would someone want to learn about this paper if they do not know about Feed-forward nn.

Comment (lemairecarl) [inline deep-learning/_posts/2017-03-27-layer-normalization.md]: It's mostly for describing the notation. It could be shortened, though.

File: deep-learning/_posts/2017-03-27-layer-normalization.md (hunk 2 above)
Comment (ppoulin91) [inline deep-learning/_posts/2017-03-27-layer-normalization.md]: I tried to stay closer to the paper's terms and notation, but it's a good point, I'll change it.

File: deep-learning/_posts/2017-03-27-layer-normalization.md (hunk 3 above)
Comment (ppoulin91) [inline deep-learning/_posts/2017-03-27-layer-normalization.md]: Exactly, the notation is necessary for the rest to make sense, but I'll drop the text and summarize it in a better way.

File: deep-learning/_posts/2017-03-27-layer-normalization.md (hunk 1 above)
Comment (ppoulin91) [inline deep-learning/_posts/2017-03-27-layer-normalization.md]: Just checked, the `layers` tag is missing from batch normalization.

**Comment by jodoin** [comment]
How come I cannot see your branch when I do a git branch -av (git pull tells me that everything is up to date)?

**Comment by Dref360** [comment]
He forked the repo

**Comment by jodoin** [comment]
Next time, can we just clone (and not fork) the repo?

**Comment by jodoin** [comment]
Nice review.

**Comment by lemairecarl** [comment]
I agree, please make a branch on our repo next time, because then it's easier to pull the branch and see the page rendered.
The last changes are good.

**Comment by ppoulin91** [comment]
Got it for the fork, I didn't think about the rendering issues... I'll clone it next time
